source/common/http/conn_manager_impl.cc

```diff
@@ -91,7 +93,11 @@ ConnectionManagerImpl::~ConnectionManagerImpl() {
     if (codec_->protocol() == Protocol::Http2) {
       stats_.named_.downstream_cx_http2_active_.dec();
     } else {
-      stats_.named_.downstream_cx_http1_active_.dec();
+      if (isWebSocketConnection()) {
+        stats_.named_.downstream_cx_websocket_active_.dec();
```

Comment:
Optimally, you would verify this stat dance in your integration tests. Can you make sure that the gauges are correct at various points?